### PR TITLE
RSDK-7754: Add SetRPM to sabertooth and fake motor drivers

### PR DESCRIPTION
**Pull request by martha-johnston**
have not tested on sabertooth motor, but I will add some software tests

edit: software tests added

### CODE REVIEW

**Review by JohnN193 — APPROVED** [review]

**Comment by martha-johnston** [comment]
@randhid thoughts on merging this to unblock Peter? I haven't been able to test with sabertooth motors yet but since it's not a super used driver I think it should be fine? Let me know if you'd like me to separate them and only merge fake

**Comment by randhid** [comment]
> @randhid thoughts on merging this to unblock Peter? I haven't been able to test with sabertooth motors yet but since it's not a super used driver I think it should be fine? Let me know if you'd like me to separate them and only merge fake

Go For it. ;)

File: components/motor/dimensionengineering/sabertooth.go [hunk 1]
```diff
@@ -422,7 +423,12 @@ func (m *Motor) GoTo(ctx context.Context, rpm, position float64, extra map[strin
 
 // SetRPM instructs the motor to move at the specified RPM indefinitely.
 func (m *Motor) SetRPM(ctx context.Context, rpm float64, extra map[string]interface{}) error {
-	return motor.NewSetRPMUnsupportedError(m.Name().ShortName())
+	if m.maxRPM == 0 {
+		return motor.NewZeroRPMError()
+	}
+
+	powerPct := rpm / m.maxRPM
```
Comment (randhid) [inline components/motor/dimensionengineering/sabertooth.go:430]: In a follow up pr, clamp this to 1 and -1.

Comment (randhid) [reply components/motor/dimensionengineering/sabertooth.go:430]: actually nvm , SetPower already clamps it, we're good!

**Pull request merged**